agent-inject/agent/container_sidecar_test.go

```diff
@@ -99,6 +99,80 @@ func TestContainerSidecarVolume(t *testing.T) {
 	)
 }
 
+func TestContainerSidecarVolumeWithIRSA(t *testing.T) {
+	
+	annotations := map[string]string{
+		AnnotationVaultRole: "foobar",
+		// this will have different mount path
+		fmt.Sprintf("%s-%s", AnnotationAgentInjectSecret, "secret1"):     "secrets/secret1",
+		fmt.Sprintf("%s-%s", AnnotationVaultSecretVolumePath, "secret1"): "/etc/container_environment",
+
+		// this secret will have same mount path as default mount path
+		// adding this so we can make sure we don't have duplicate
+		// volume mounts
+		fmt.Sprintf("%s-%s", AnnotationAgentInjectSecret, "secret2"):     "secret/secret2",
+		fmt.Sprintf("%s-%s", AnnotationVaultSecretVolumePath, "secret2"): "/etc/default_path",
+
+		// Default path for all secrets
+		AnnotationVaultSecretVolumePath: "/etc/default_path",
+
+		fmt.Sprintf("%s-%s", AnnotationAgentInjectSecret, "secret3"): "secret/secret3",
+	}
+
+	pod := testPodIRSA(annotations)
+	var patches []*jsonpatch.JsonPatchOperation
+
+	err := Init(pod, AgentConfig{"foobar-image", "http://foobar:1234", "aws", "test", "test", true, "1000", "100", DefaultAgentRunAsSameUser, DefaultAgentSetSecurityContext, ""})
+	if err != nil {
+		t.Errorf("got error, shouldn't have: %s", err)
+	}
+
+	agent, err := New(pod, patches)
+	agent.AwsIamTokenAccountName = "aws-iam-token"
+	agent.AwsIamTokenAccountPath = "/var/run/secrets/eks.amazonaws.com/serviceaccount"
+
+	if err := agent.Validate(); err != nil {
+		t.Errorf("agent validation failed, it shouldn't have: %s", err)
+	}
+
+	container, err := agent.ContainerSidecar()
+
+	// One token volume mount, one config volume mount and two secrets volume mounts
+	require.Equal(t, 5, len(container.VolumeMounts))
+
```

Comment:
The require.Equal on the next line will also check the length, so I think this check is not needed.

```suggestion
```